include/mata/nfa.hh

```diff
@@ -271,6 +315,44 @@ public:
         return post[q];
     };
 
+    // TODO: why do we have the code of all these methods in the header file? Should we move it out?
+    const std::vector<State> defragment(Util::NumberPredicate<State> & is_staying) {
+
+        //first, indexes of post are filtered (places of to be removed states are taken by states on their right)
+        Util::filter_indexes(post, [is_staying](State i) { return is_staying[i]; });
+
+        //get renaming of current states to new numbers:
+        std::vector<State> renaming(this->find_max_state()+1);
+        size_t i = 0;
+        for (State q:is_staying.get_elements()) {
+            if (q >= renaming.size())
+                break;
+            renaming[q] = i;
```

Comment:
I think, this can be shortened to `renaming[q] = i++;`

Reply:
It is an evil code, I don't want to touch it